l10n_nl_cbs_export/models/cbs_export_file.py

```diff
@@ -0,0 +1,146 @@
+# -*- coding: utf-8 -*-
+# See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models, api, _
+import base64
+from datetime import datetime ,date ,timedelta
+from odoo.exceptions import ValidationError
+import calendar
+import time
+
+
+class cbs_export_file(models.Model):
+    _name = 'cbs.export.file'
+    _inherit = ['mail.thread', 'ir.needaction_mixin']
+
+    @api.multi
+    def get_filename(self):
+        for rec in self:
+            rec.filename = '%s_%s.csv' % (rec.month, rec.year)
+
+    cbs_export_invoice = fields.Binary(string='CBS Export File', attachment=True)
+    filename = fields.Char(string='Filename', size=64, compute='get_filename')
+    name = fields.Char(string='Name', size=64)
+    month = fields.Selection([
+        ('01', 'January'),
+        ('02', 'February'),
+        ('03', 'March'),
+        ('04', 'April'),
+        ('05', 'May'),
+        ('06', 'June'),
+        ('07', 'July'),
+        ('08', 'August'),
+        ('09', 'September'),
+        (10, 'October'),
+        (11, 'November'),
+        (12, 'December')
+    ], default='01', string='Month')
+    year = fields.Char('Year')
+    account_invoice_ids = fields.One2many('account.invoice', 'cbs_export_id')
+
+    @api.model
+    def create(self, vals):
+        vals['name'] = self.env['ir.sequence'].next_by_code('cbs.export.file')
+        return super(cbs_export_file, self).create(vals)
+
+    @api.one
+    def get_data(self):
+        self.set_invoice()
+        self.export_file()
+
+    @api.constrains('year')
+    def check_year(self):
+        if self.year:
+            if not self.year.isdigit():
+                raise ValidationError(_("Please insert a valid Year"))
+            else:
+                is_valid_year = '%d' % (int(self.year))
+                try:
+                    time.strptime(is_valid_year, '%Y')
+                except ValueError:
+                    raise ValidationError(_("Please insert a valid Year"))
+
+    @api.one
+    def set_invoice(self):
+        days = calendar.monthrange(int(self.year), int(self.month))
+        account_ids = self.env['account.invoice'].search([
```

Comment:
`account_ids` is a very misleading name for a variable holding a recordset of invoices